src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileRepository.cs

```diff
@@ -24,20 +24,27 @@ public FileRepository(ContentDbContext contentDbContext)
             string filename,
             FileType type,
             File replacingFile = null,
-            File source = null)
+            File source = null,
+            Guid? subjectId = null)
         {
             if (type == DataZip)
             {
                 throw new ArgumentException($"Cannot use generic Create method for type {DataZip}",
                     nameof(type));
             }
 
+            if ((type == FileType.Data || type == Metadata) && subjectId == null)
```

Comment:
I think you should make a separate method for creating a data/meta file which has the `replacingFile`, `source` and `subjectId` fields, where `subjectId` will be non-optional. It will be much clearer then which fields are expected for ancillary/chart fields (always only `releaseId, `filename` and `type`), and which fields are expected when creating data/meta files.